charts/rancher-monitoring/v0.0.4/templates/_helpers.tpl

```diff
@@ -136,4 +136,12 @@ add below linux tolerations to workloads could be scheduled to those linux nodes
   value: "linux"
   effect: "NoSchedule"
   operator: "Equal"
-{{- end -}}
\ No newline at end of file
+{{- end -}}
+
+{{- define "linux-node-selector" -}}
+{{- if semverCompare "<1.14-0" .Capabilities.KubeVersion.GitVersion -}}
```

Comment:
Because for windows cluster, we need to schedule all the monitoring components to Linux nodes, so we need `nodeSelector`. After k8s 1.14, label `beta.kubernetes.io/os` is deprecated, use the label `kubernetes.io/os` instead. When a user has a k8s 1.13 cluster with monitoring enabled, and upgrade server to latest, then user could upgrade monitoring, if we only have `kubernetes.io/os` label, it will schedule fail. So for old version cluster, we can use the label `beta.kubernetes.io/os` or empty nodeSelector, I prefer the first way, in this way only kubeVersion affect nodeSelector. Even we not only support k8s 1.15, the monitoring still work well, still could be scheduled to Linux nodes.